plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -346,6 +362,32 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         return legacyToWriteMapping(session, type);
     }
 
+    private LongWriteFunction sqlServerTimeWriteFunction(int precision)
+    {
+        return new LongWriteFunction()
+        {
+            @Override
+            public String getBindExpression()
+            {
+                // Binding setObject(LocalTime) can result with "The data types time and datetime are incompatible in the equal to operator."
+                // when write function is used for predicate pushdown.
+                return format("CAST(? AS time(%s))", min(precision, MAX_SUPPORTED_TIMESTAMP_PRECISION));
+            }
+
+            @Override
+            public void set(PreparedStatement statement, int index, long picosOfDay)
+                    throws SQLException
+            {
+                picosOfDay = round(picosOfDay, 12 - precision);
```

Comment:
I wish there were static methods with helpful names to do the rounding for us (handling overflow too).

No changes requested.